src/websocket.ts

```diff
@@ -36,9 +39,10 @@ export default class WS {
     WS.instances = [];
   }
 
-  constructor(url: string, { jsonProtocol = false }: WSOptions = {}) {
+  constructor(url: string, opts: WSOptions = {}) {
     WS.instances.push(this);
 
+    const { jsonProtocol, ...serverOptions } = opts;
```

Comment:
This should be something like:

```suggestion
    const { jsonProtocol = false, ...serverOptions } = opts;
```